test/jdk/java/net/httpclient/HttpResponseInputStreamInterruptTest.java

```diff
@@ -0,0 +1,171 @@
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8294047
+ * @library /test/lib
+ * @run junit HttpResponseInputStreamInterruptTest
+ */
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+import jdk.test.lib.net.URIBuilder;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.concurrent.CountDownLatch;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class HttpResponseInputStreamInterruptTest {
+
+    HttpServer server;
+    int port;
+    private CountDownLatch countDownLatch = new CountDownLatch(2);
+    private CountDownLatch interruptLatch = new CountDownLatch(1);
+    static final String FIRST_MESSAGE = "Should be received";
+    static final String SECOND_MESSAGE = "Shouldn't be received";
+
+    @BeforeAll
+    void before() throws Exception {
+        InetAddress loopback = InetAddress.getLoopbackAddress();
+        InetSocketAddress addr = new InetSocketAddress(loopback, 0);
+        server = HttpServer.create(addr, 0);
+        port = server.getAddress().getPort();
+        Handler handler = new Handler(countDownLatch, interruptLatch);
+        server.createContext("/", handler);
+        server.start();
+    }
+
+    @AfterAll
+    void after() throws Exception {
+        server.stop(0);
+    }
+
+    @Test
+    public void test() {
+        try {
+            //create client and interrupter threads
+            Thread clientThread = createClientThread(countDownLatch, port);
+            Thread interrupterThread = new Thread(() -> {
+                try {
+                    countDownLatch.await();
+                    clientThread.interrupt();
+                    interruptLatch.countDown();
+                } catch (InterruptedException e) {
+                    System.out.println("interrupterThread failed");
+                    throw new RuntimeException(e);
+                }
+            });
+
+            // Start the threads then wait until clientThread completes
+            clientThread.start();
+            interrupterThread.start();
+            clientThread.join();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    static class Handler implements HttpHandler {
+
+        CountDownLatch countDownLatch;
```

Comment:
Maybe a better name than `countDownLatch` could be found - that could give a clue on what this latch is used to wait for?

Reply:
I was thinking maybe `messageLatch`, `messageReceivedLatch` or `clientReadyLatch`

Reply:
Maybe we rename both:
```
countDownLatch  => interruptReadyLatch
interruptLatch  => interruptDoneLatch
```